src/esphomelib/time/rtc_component.h

```diff
@@ -0,0 +1,58 @@
+//
+//  rtc_component.h
+//  esphomelib
+//
+//  Created by Brad Davidson on 06.06.18.
+//  Copyright © 2018 Otto Winter. All rights reserved.
+//
+
+#ifndef ESPHOMELIB_TIME_RTC_COMPONENT_H
+#define ESPHOMELIB_TIME_RTC_COMPONENT_H
+
+
+#include "esphomelib/component.h"
+#include "esphomelib/defines.h"
+#include <stdlib.h>
+#include <time.h>
+
+#ifdef USE_RTC_COMPONENT
+
+ESPHOMELIB_NAMESPACE_BEGIN
+
+namespace time {
+
+/// The RTC component exposes common timekeeping functions via the device's local real-time clock.
+///
+/// \note
+/// Note that the ESP's C library only supports TZ strings that specify an offset and DST info;
+/// you cannot specify zone names or paths to zoneinfo files.
+/// \see https://www.gnu.org/software/libc/manual/html_node/TZ-Variable.html
+class RTCComponent : public Component {
+ public:
+  explicit RTCComponent(const std::string &tz);
+
+  void setup() override;
+
+  void set_timezone(const std::string &tz);
+
+  std::string get_timezone();
+
+  time_t time(time_t *_timer);
```

Comment:
* These methods should be documented. Also their names are not too great (I know they're from libc, but we don't have to repeat their mistakes). 
* Also, concerning the API, the user shouldn't need to do this:

```cpp
auto time = rtccomponent->time(nullptr);
auto tm = rtccomponent->localtime(&time);
``

I think the `time` method should be dropped and `localtime` should call `time` internally. Also, let's not return a pointer below. lambda users shouldn't need to learn the whole arrow operator syntax. The C standard library's time.h is just a mess and please let's not replicate that here.

```cpp
// Pseudocode
struct EsphomelibTime {
  uint8_t seconds;
  uint8_t minutes;
  uint8_t hours;
  uint8_t day_of_week;
  uint8_t day_of_month;
  uint16_t day_of_year;
  uint8_t month;
  uint16_t year;
}

// ...

// I'm calling them like python's datetime methods here to avoid confusion with libc's API
// Few people will call the utcnow() method anyway, as for most use cases I think now would make
// much more sense (of course for other software this would be different).
EsphomelibTime utcnow() {
  time_t *t = ::time(nullptr);
  struct tm *c_time = ::gmtime(t);
  return prettify_time_(c_time);
}

EsphomelibTime now() {
  time_t *t = ::time(nullptr);
  struct tm *c_time = ::localtime(t);
  return prettify_time_(c_time);
}



EsphomelibTime prettify_time_(struct tm *c_time) {
  return EsphomelibTime {
    .seconds = c_time->tm_sec,
	.minutes = c_time->tm_min,
	.hours = c_time->tm_hour,
	.day_of_week  = c_time->tm_wday,
	.day_of_month = c_time->tm_mday,
	.day_of_year = c_time->tm_yday,
	.month = c_time->tm_mon,
	.year = 1900 + c_time->tm_year,
  };
}
```

Problem then of course is how to handle `strftime`, but I think we can just re-convert then.